pypsa/components/descriptors.py

```diff
@@ -117,11 +118,32 @@ def get_activity_mask(
     return mask
 
 
+@overload
 def get_bounds_pu(
     c: Components,
     sns: Sequence,
     index: pd.Index | None = None,
     attr: str | None = None,
+    as_xarray: Literal[True] = True,
+) -> tuple[xr.DataArray, xr.DataArray]: ...
+
+
+@overload
+def get_bounds_pu(
+    c: Components,
+    sns: Sequence,
+    index: pd.Index | None = None,
+    attr: str | None = None,
+    as_xarray: Literal[False] = False,
+) -> tuple[pd.DataFrame, pd.DataFrame]: ...
+
+
+def get_bounds_pu(
+    c: Components,
+    sns: Sequence,
+    index: pd.Index | None = None,
+    attr: str | None = None,
+    as_xarray: bool = False,
```

Comment:
I am still thinking that this should also be moved to the component classes in order to avoid the if cases to differentiate between components here